pkg/graveler/graveler.go

```diff
@@ -3050,6 +3056,15 @@ func (g *Graveler) DiffUncommitted(ctx context.Context, repository *RepositoryRe
 			return nil, err
 		}
 	}
+	if branch.BaseMetaRangeID != "" {
+		compactedDiffIterator, err := g.CommittedManager.Diff(ctx, repository.StorageNamespace, metaRangeID, branch.BaseMetaRangeID)
+		if err != nil {
+			valueIterator.Close()
+			committedValueIterator.Close()
+			return nil, err
+		}
+		return NewCombinedDiffIterator(compactedDiffIterator, committedValueIterator, valueIterator), nil
```

Comment:
I read CombinedDiffIterator docs (below) 3 times and I'm still not sure the usage of it here is correct. Can you elaborate on why this is true?

```
// CombinedDiffIterator calculates the diff between a commit and a branch, including the staging area of the branch.
// committedDiffIterator is the DiffIterator between the commit and the HEAD of the branch.
// stagingIterator is the ValueIterator on the staging area of the branch
// leftIterator is the ValueIterator on the commit
```